torch/_subclasses/fake_tensor.py

```diff
@@ -154,6 +155,16 @@ def _is_tensor_constructor(func: OpOverload):
     )
 
 
+def is_fake(x):
+    if isinstance(x, FakeTensor):
+        return True
+    if is_traceable_wrapper_subclass(x):
+        flattened_tensors, _ = type(x).__tensor_flatten__(x)
+        # need to recurse because we could have nested subclasses
+        return all(is_fake(x) for x in flattened_tensors)
```

Comment:
What is the invariant here? When is it possible for this to be partially fake?

Reply:
Good question - we should probably assert here. Something's probably gone horribly wrong if we have a wrapper subclass that holds some real tensors and some fake tensors.